ydb/library/yql/providers/yt/provider/yql_yt_join_reorder.cpp

```diff
@@ -66,50 +66,40 @@ class TJoinReorderer {
     }
 
     TYtJoinNodeOp::TPtr Do() {
-        CollectRels(Root);
-        if (!CollectOps(Root)) {
-            return Root;
-        }
-
-        IOptimizer::TInput input;
-        input.EqClasses = std::move(EqClasses);
-        input.Left = std::move(Left);
-        input.Right = std::move(Right);
-        input.Rels = std::move(Rels);
-        input.Normalize();
-        YQL_CLOG(INFO, ProviderYt) << "Input: " << input.ToString();
-
-        std::function<void(const TString& str)> log;
+        std::shared_ptr<IBaseOptimizerNode> tree;
+        std::shared_ptr<IProviderContext> ctx;
+        BuildOptimizerJoinTree(tree, ctx, Root);
 
-        log = [](const TString& str) {
-            YQL_CLOG(INFO, ProviderYt) << str;
-        };
-
-        std::unique_ptr<IOptimizer> opt;
+        std::unique_ptr<IOptimizerNew> opt;
 
         switch (State->Types->CostBasedOptimizer) {
-        case ECostBasedOptimizerType::PG:
-            opt = std::unique_ptr<IOptimizer>(MakePgOptimizer(input, log));
-            break;
+//        case ECostBasedOptimizerType::PG:
+//            opt = std::unique_ptr<IOptimizer>(MakePgOptimizer(input, log));
+//            break;
         case ECostBasedOptimizerType::Native:
-            opt = std::unique_ptr<IOptimizer>(NDq::MakeNativeOptimizer(input, log));
+            opt = std::unique_ptr<IOptimizerNew>(NDq::MakeNativeOptimizerNew(*ctx, 100000));
```

Comment:
Do you really want to plug in a magic number here? Why not create a pragma for this?
We have a pragma in DG for this parameter

Reply:
In next PR maybe.